storage/google/cloud/storage/client.py

```diff
@@ -309,6 +311,31 @@ def create_bucket(self, bucket_or_name, requester_pays=None, project=None):
         bucket.create(client=self, project=project)
         return bucket
 
+    def download_blob_to_file(self, blob_or_uri, file_obj, start=None, end=None):
+        """Download the contents of a blob object or blob URI into a file-like object.
+
+        Args:
+            blob_or_uri (Union[ \
+                :class:`~google.cloud.storage.blob.Blob`, \
+                 str, \
+            ]):
+                The blob resource to pass or URI to download.
+            file_obj (file):
+                A file handle to which to write the blob's data.
+            start (int):
+                Optional, the first byte in a range to be downloaded.
+            end (int):
+                Optional, The last byte in a range to be downloaded.
+        """
+        try:
+            blob_or_uri.download_to_file(file_obj, client=self, start=start, end=end)
+        except AttributeError:
+            scheme, netloc, path, query, frag = urlsplit(blob_or_uri)
+            bucket = Bucket(self, name=netloc)
```

Comment:
Let's raise a `ValueError` if `scheme` isn't `gs`. I was warned that people might start passing in `http` URLs and expect this to work.

Reply:
We'll need a test for this check. (Try a `https://` link and verify that `ValueError` is raised with `pytest.raises`

Reply:
I added an `if` statement for raising `ValueError` if scheme isn't `gs` as directed. I also added the test for the change.